opensearch_py_ml/semantic search/train.py

```diff
@@ -0,0 +1,358 @@
+# SPDX-License-Identifier: Apache-2.0
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+# Any modifications Copyright OpenSearch Contributors. See
+# GitHub history for details.
+
+import argparse
+import os
+import pickle
+import random
+import shutil
+import time
+from typing import List
+from zipfile import ZipFile
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import torch
+from sentence_transformers import InputExample, SentenceTransformer
+from tqdm import tqdm
+from transformers import get_linear_schedule_with_warmup
+
+
+def train_semantic_search_model(
```

Comment:
I'm not expert, this is NOT a suggestion, just my thought: should we use class style coding? For Java we always create class to wrap function and fields. I know python could be function style or class style, for example [SentenceTransformer](https://github.com/UKPLab/sentence-transformers/blob/master/sentence_transformers/SentenceTransformer.py#L33), they are using class style.  I guess there will be some pros/cons between these two coding styles. Again, just curious and put my thought here.

Reply:
converted the class in the new commit